app/views/samples/_index_bulk_actions_js.haml

```diff
@@ -22,3 +28,15 @@
   function selectedCheckboxes() {
     return allCheckboxes().filter(":checked");
   }
+
+  function selectedSamplesIds() {
+    var samples = $('.laboratory-sample-row');
+    return $.map(samples, function(sample) {
+      var checkboxInput = $(sample).find('input[name="sample_ids[]"]');
+      if(checkboxInput.is(':checked')) {
+        var uuid = $(sample).children('#sample_uuid').text();
+        var isolateName = $(sample).children('#sample_isolate_name').text();
+        return {uuid, isolateName};
+      }
```

Comment:
Hmm, this function returns an array of the selected samples only. If the user selects two samples, the function returns an array of two elements. Not sure if this is what you mean or I am missing something.

Reply:
What I meant is that the `map` callback does return a `{uuid, isolateName}` object if the checkbox is checked, but when it's not checked, what happens? It will return `undefined` (I got mistaken with the function returning `null`, it's JS so it's `undefined` :facepalm:). For example:

```js
[1, 2, 3, 4, 5].map((x) => {
  if (x > 2) {
    return x
  }
})
// => [undefined, undefined, 3, 4, 5]
```

I wanted to highlight this. Are those `undefined` handled somewhere?